source/dagre.js

```diff
@@ -1591,7 +1591,7 @@ dagre.layout = (graph, layout) => {
                             }
                             const label = g.node(v).label;
                             if (min !== Number.POSITIVE_INFINITY && label.borderType !== borderType) {
-                                xs[v] = Math.max(xs[v], min);
+                                xs[v] = Math.min(xs[v], min);
```

Comment:
this will choose smallest width

Reply:
@seanshpark @lutzroeder 

How about the following code?

```js
                            let minOut = Number.POSITIVE_INFINITY;
                            let minIn = Number.POSITIVE_INFINITY;
                            for (const e of blockG.node(v).out) {
                                minOut = Math.min(minOut, xs[e.w] - e.label);
                            }
                            for (const e of blockG.node(v).in) {
                                minIn = Math.min(minIn, xs[e.w] - e.label);
                            }
                            const label = g.node(v).label;
                            if (minOut !== Number.POSITIVE_INFINITY && minIn !== Number.NEGATIVE_INFINITY && label.borderType !== borderType) {
                                if (Math.abs(xs[v] - minOut) < Math.abs(xs[v] - minIn)) {
                                    xs[v] = Math.max(xs[v], minOut);
                                }else {
                                    xs[v] = Math.max(xs[v], minIn);
                                }
                            }
```

model | before | after
------|--------|-------
yolov5n | ![yolov5n onnx](https://github.com/lutzroeder/netron/assets/1131125/5b930f8d-32fd-424e-bc2e-194d4bd0295e) | ![new_yolov5n onnx](https://github.com/lutzroeder/netron/assets/1131125/6aa6b9f4-6ace-4b9c-bdf0-66e375abef46)
sample.zip | ![saved_model pbtxt](https://github.com/lutzroeder/netron/assets/1131125/165387da-6e5a-470f-8adf-753f5e6469df) | ![new_saved_model pbtxt](https://github.com/lutzroeder/netron/assets/1131125/9d804122-2cb2-4e64-8052-ed61a67d1090)

Reply:
> How about the following code?

This looks better !